nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp

```diff
@@ -361,9 +361,61 @@ CComPtr<IAccessible2> getTextBoxInComboBox(
 	return child;
 }
 
+
+void fillRole(IAccessible2* pacc, VARIANT* varChild, long* role, BSTR* roleString) {
+	if (pacc->role(role) != S_OK) {
+		*role = IA2_ROLE_UNKNOWN;
+	}
+	VARIANT varRole;
+	VariantInit(&varRole);
+	if (*role == 0) {
+		if (pacc->get_accRole(*varChild, &varRole) != S_OK) {
+			LOG_DEBUG(L"accRole failed");
+		}
+		if (varRole.vt == VT_I4) {
+			*role = varRole.lVal;
+		}
+		else if (varRole.vt == VT_BSTR) {
+			*roleString = varRole.bstrVal;
+		}
+	}
+	VariantClear(&varRole);
+}

```

Comment:
This should be using RAII types for `pacc, varChild, roleString, varRole`. 
See `appendNameDescription` in `nvdaHelper/remote/textFromIAccessible.cpp` for examples.

Rather than taking pointers, return a `std::tuple` to make use of auto unpacking https://en.cppreference.com/w/cpp/language/structured_binding

Reply:
I can't figure out how to do this for the `VARIANT` type `varChild`. Same with `VBufStorage_buffer_t* buffer, VBufStorage_controlFieldNode_t* parentNode` in https://github.com/nvaccess/nvda/pull/13649#discussion_r874758216

Reply:
We can talk about this tomorrow.